onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -62,6 +62,19 @@ X --> ReduceMean --> Sub -->  Pow --> ReduceMean --> Add --> Sqrt --> Div --> Mu
                       |                                                ^
                       |                                                |
                       +------------------------------------------------+
+
+When using Apex O2, a Cast node may be inserted between Div and Mul, Layer Normalization will also handle the case below:
++---------------------+
+|                     |
+|                     v
+X --> ReduceMean --> Sub --> Pow --> ReduceMean --> Add --> Sqrt --> Div --> Cast --> Mul --> Add
+                      |                                               ^
+                      |                                               |
+                      +-----------------------------------------------+
+
+Since LayerNormalization supports input and scale/bias have different data type, and during the kernel execution,
```

Comment:
nit: "have different data type" -> 'in different data types'

Reply:
Can you help name this new pattern differently, for example "Apex O2", for the blocks that are specific for this pattern, we add comment 

"Apex O2 pattern specific match starts"
...
"Apex O2 pattern specific  match ends"

Once the change got merged, it's really hard to understand which piece of code is doing what pattern match....